app/pages/project/instances/instance/InstancePage.tsx

```diff
@@ -94,7 +94,10 @@ export function InstancePage() {
   const makeActions = useMakeInstanceActions(instanceSelector, {
     onSuccess: refreshData,
     // go to project instances list since there's no more instance
-    onDelete: () => navigate(pb.instances(instanceSelector)),
+    onDelete: () => {
+      apiQueryClient.invalidateQueries('instanceList')
+      navigate(pb.instances(instanceSelector))
+    },
```

Comment:
This was a legit bug, and I know why this change turned it up. The failure was that the expected disappearance of `you-fail` from the instance list was not happening.

What happens in this instance delete test is 

1. go to the instance detail page for `you-fail`
2. delete it
3. let it automatically land us on the list view
4. confirm the instance is gone
 
Before this PR, there was nothing on the instance detail page "observing" the `instanceList` query, so when we nav to instance list on delete success, that query will always be fetched fresh because it is "new" from RQ's point of view. But with this PR, there is now the breadcrumb that uses the list of instances, so the query is "active" on the instance detail page.

Here is the funny bit: when you do what the test does _by hand_, you inevitably sit on the `you-fail` page longer than the 2 seconds required for the query to become stale. That means it will automatically refetch when the loader fires for the instance list page on navigation. But the test is faster than you, so it lands on the instance detail and hits delete in under 2 seconds, so the query is _not_ stale by the time we land on the instance list and therefore does not refetch. I confirmed this theory by adding a sleep to the test so it sits on the page for 2 seconds before clicking delete. That makes the test pass! But of course the real fix is to specifically invalidate the instance list in the delete success callback.

Reply:
Oh, wow. Good sleuthing on this.